SingularityService/src/main/java/com/hubspot/singularity/data/StateManager.java

```diff
@@ -312,6 +313,20 @@ private boolean requestTypeIsOnDemand(SingularityPendingTaskId taskId) {
     return false;
   }
 
+  private List<SingularityPendingTaskId> getOnDemandLateTasks (List<SingularityPendingTaskId> maybeOnDemandLateTasks) {
+    List<SingularityPendingTaskId> onDemandLateTasks = new ArrayList<>();
+    for (SingularityPendingTaskId maybeOnDemandLateTask : maybeOnDemandLateTasks) {
+      String requestId = requestManager.getRequest(maybeOnDemandLateTask.getRequestId()).get().getRequest().getId();
+      Optional<Integer> maybeInstancesLimit = requestManager.getRequest(maybeOnDemandLateTask.getRequestId()).get().getRequest().getInstances();
```

Comment:
since `requestManager.getRequest` returns an Optional, we should have a check for isPresent here to avoid throwing an exception up the stack. e.g.

Optional<SingularityRequest> maybeRequest = requestManager.getRequest(...)
if (maybeRequest.isPresent) { continue to do stuff }

Reply:
Got the `maybeOnDemandLateTasks` from `lateTasksPartitionedByOnDemand` which calls `requestTypeIsOnDemand` that had already checked that the request is optional...checking again here will be repetitive